app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\{
+    User,
+    Authentication
+};
+use App\Enums\AuthenticationStatus;
+use App\Http\Requests\UserRequest;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Hash;
+use Exception;
+use Carbon\Carbon;
+
+class UserController extends Controller
+{
+    public function create(Request $request)
+    {
+        $user_token = $request->authentication_token;
+        $authentication = Authentication::where('token', $user_token)
+            ->where('status', AuthenticationStatus::MAIL_SENT)
+            ->where('expired_at', '>', Carbon::now())
+            ->first();
+
+        if(is_null($authentication)){
+            return to_route('tasks.index')->withErrors(['status_error' => '会員登録に失敗しました。']);
+        }
+        return view('user.create');
+    }
+
+    public function store(UserRequest $request)
+    {   
+        $user_token = $request->authentication_token;
+        $authentication = Authentication::where('token', $user_token)
+            ->where('status', AuthenticationStatus::MAIL_SENT)
+            ->where('expired_at', '>', Carbon::now())
+            ->first();
+        if(is_null($authentication)){
+            return to_route('tasks.index')->withErrors(['status_error' => '会員登録に失敗しました。']);
+        }
+        
+        try{
+            User::create([
+                'email' => $authentication['email'],
```

Comment:
```suggestion
                'email' => $authentication->email,
```
モデルのプロパティの呼び出しは矢印を使って書いてください。
大括弧を書くやり方は連想配列から値をキーで呼び出す場合の書き方なので、$authenticationが配列型の変数なのかと思われてしまうからです。

Reply:
ご指摘、ご説明ありがとうございます。
プロパティの取得方法を修正致しました。
ご確認宜しくお願い致します。
app/Http/Controllers/UserController.php